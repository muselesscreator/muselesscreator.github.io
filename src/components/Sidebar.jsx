import React from 'react';
import classNames from 'classnames';
// import { Image } from 'react-bootstrap';
import {
  FaEnvelope,
  FaLinkedin,
  FaChevronDown,
  FaGithub,
} from 'react-icons/fa';
import { FaPhone } from 'react-icons/fa6';
import contactInfo from '../data/contact';

const Sidebar = () => {
  const [isActive, setIsActive] = React.useState(false);
  const toggle = () => setIsActive(!isActive);
  return (
    <aside className={classNames('sidebar', { active: isActive })}>
      <div className="sidebar-info">
        <figure className="avatar-box">
          <img src={contactInfo.avatar} alt={contactInfo.name} width="80" />
        </figure>
        <div className="info-content">
          <h1 className="name" title={contactInfo.name}>{contactInfo.name}</h1>
          <p className="title">{contactInfo.title}</p>
        </div>
        <button onClick={toggle} className="info_more-btn">
          <span>Show Contacts </span>
          <FaChevronDown />
        </button>
      </div>
      <div className="sidebar-info_more">
        <div className="separator" />
        <ul className="contacts-list">
          <li className="contact-item">
            <div className="icon-box">
              <FaEnvelope />
            </div>
            <div className="contact-info">
              <p className="contact-title">Email</p>
              <a href={`mailto:${contactInfo.email}`} className="contact-link">{contactInfo.email}</a>
            </div>
          </li>
          <li className="contact-item">
            <div className="icon-box">
              <FaPhone />
            </div>
            <div className="contact-info">
              <p className="contact-title">Phone</p>
              <a href={`tel:${contactInfo.phoneRaw}`}  className="contact-link">{contactInfo.phone}</a>
            </div>
          </li>
          <li className="contact-item">
            <div className="icon-box">
              <FaGithub />
            </div>
            <div className="contact-info">
              <p className="contact-title">Github</p>
              <a href={contactInfo.github}  className="contact-link">{contactInfo.github}</a>
            </div>
          </li>
          <li className="contact-item">
            <div className="icon-box">
              <FaLinkedin />
            </div>
            <div className="contact-info">
              <p className="contact-title">LinkedIn</p>
              <a href={contactInfo.linkedIn}  className="contact-link">{contactInfo.linkedIn}</a>
            </div>
          </li>
        </ul>
      </div>
    </aside>
  );
};

export default Sidebar;
