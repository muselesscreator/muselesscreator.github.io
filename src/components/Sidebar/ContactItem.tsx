export type ContactItemProps = {
  IconEl: React.FC;
  title: string;
  href: string;
  value: string;
};
const ContactItem = ({
  IconEl,
  title,
  href,
  value
}: ContactItemProps) => (
  <li className="contact-item">
    <div className="icon-box">
      <IconEl />
    </div>
    <div className="contact-info">
      <p className="contact-title">{title}</p>
      <a href={href} className="contact-link">{value}</a>
    </div>
  </li>
);

export default ContactItem;
