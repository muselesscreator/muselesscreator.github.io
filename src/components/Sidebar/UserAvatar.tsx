import contactInfo from '~/data/contact';

const UserAvatar = () => (
  <figure className="avatar-box">
    <img src={contactInfo.avatar} alt={contactInfo.name} width="80" />
  </figure>
);

export default UserAvatar;
