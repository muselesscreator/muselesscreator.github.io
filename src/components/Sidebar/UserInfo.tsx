import contactInfo from '~/data/contact';

const UserInfo = () => (
  <div className="info-content">
    <h1 className="name" title={contactInfo.name}>{contactInfo.name}</h1>
    <p className="title">{contactInfo.title}</p>
  </div>
);

export default UserInfo;
