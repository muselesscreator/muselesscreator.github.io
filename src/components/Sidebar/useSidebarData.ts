import React from 'react';

const useSidebarData = () => {
  const [isActive, setIsActive] = React.useState(false);
  const toggle = () => setIsActive(!isActive);
  return {
    isActive,
    toggle,
  };
};

export default useSidebarData;
