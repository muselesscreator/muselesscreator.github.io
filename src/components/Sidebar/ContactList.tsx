import {
  FaEnvelope,
  FaLinkedin,
  FaGithub,
} from 'react-icons/fa';
import { FaPhone } from 'react-icons/fa6';

import { ContactInfo } from '~/data/contact';

import ContactItem from './ContactItem';

type ContactListProps = {
  contactInfo: ContactInfo;
}

const ContactList = ({ contactInfo }: ContactListProps) => {
  const fields = [
    {
      icon: <FaEnvelope />,
      title: 'Email',
      value: contactInfo.email,
      href: `mailto:${contactInfo.email}`,
    },
    {
      icon: <FaPhone />,
      title: 'Phone',
      value: contactInfo.phone,
      href: `tel:${contactInfo.phoneRaw}`,
    },
    {
      icon: <FaGithub />,
      title: 'Github',
      value: contactInfo.github,
      href: contactInfo.github,
    },
    {
      icon: <FaLinkedin />,
      title: 'LinkedIn',
      value: contactInfo.linkedIn,
      href: contactInfo.linkedIn,
    },
  ];

  return (
    <ul className="contacts-list">
      {fields.map((field) => (
        <ContactItem key={field.title} {...field} />
      ))}
    </ul>
  );
};

export default ContactList;
