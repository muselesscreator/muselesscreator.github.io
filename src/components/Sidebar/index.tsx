import classNames from 'classnames';
import { FaChevronDown } from 'react-icons/fa';

import useSidebarData from './useSidebarData';
import ContactList from './ContactList';
import UserAvatar from './UserAvatar';
import UserInfo from './UserInfo';

const Sidebar = () => {
  const { isActive, toggle } = useSidebarData();
  return (
    <aside className={classNames('sidebar', { active: isActive })}>
      <div className="sidebar-info">
        <UserAvatar />
        <UserInfo />
        <button onClick={toggle} className="info_more-btn">
          <span>Show Contacts </span>
          <FaChevronDown />
        </button>
      </div>
      <div className="sidebar-info_more">
        <div className="separator" />
        <ContactList />
      </div>
    </aside>
  );
};

export default Sidebar;
