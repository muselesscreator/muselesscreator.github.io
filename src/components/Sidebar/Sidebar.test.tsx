import {
  vi,
  describe,
  test,
  expect,
  Mock,
} from 'vitest';

import { shallow, setMockName } from '@muselesscreator/react-shallow-snapshot';

import useSidebarData from './useSidebarData';
import Sidebar from '.';

vi.mock('./useSidebarData', () => ({ default: vi.fn() }));
vi.mock('./ContactList', () => ({ default: 'ContactList' }));
vi.mock('./UserAvatar', () => ({ default: 'UserAvatar' }));
vi.mock('./UserInfo', () => ({ default: 'UserInfo' }));
vi.mock('react-icons/fa', () => ({ FaChevronDown: 'FaChevronDownIcon' }));

const hookProps = {
  isActive: true,
  toggle: vi.fn(),
};

setMockName(hookProps.toggle, 'hookProps.toggle');
(useSidebarData as Mock).mockReturnValue(hookProps);

describe('Sidebar component', () => {
  test('snapshot', () => {
    const el = shallow(<Sidebar />);
    expect(el.snapshot).toMatchSnapshot();
  });
  test('sidebar renders UserAvatar and UserInfo in sidebar-info with toggle button', () => {
    const el = shallow(<Sidebar />);
    const sidebarInfo = el.instance.findByClassName('sidebar-info')[0];
    expect(sidebarInfo.findByType('UserAvatar')).toBeTruthy();
    expect(sidebarInfo.findByType('UserInfo')).toBeTruthy();
    const button = sidebarInfo.findByType('button')[0];
    expect(button.props.onClick).toEqual(hookProps.toggle);
  });
  test('sidebar renders ContactList in info_more div', () => {
    const el = shallow(<Sidebar />);
    const infoMore = el.instance.findByClassName('sidebar-info_more')[0];
    expect(infoMore.findByType('ContactList')).toBeTruthy();
  });
});
