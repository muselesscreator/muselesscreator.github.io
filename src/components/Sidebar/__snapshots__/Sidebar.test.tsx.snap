// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sidebar component > snapshot 1`] = `
"<aside
  className="sidebar active"
>
  <div
    className="sidebar-info"
  >
    <UserAvatar />
    <UserInfo />
    <button
      className="info_more-btn"
      onClick={[Function hookProps.toggle]}
    >
      <span>
        Show Contacts 
      </span>
      <FaChevronDownIcon />
    </button>
  </div>
  <div
    className="sidebar-info_more"
  >
    <div
      className="separator"
    />
    <ContactList />
  </div>
</aside>"
`;

exports[`Sidebar component > snapshot includes user name and title 1`] = `
"<aside
  className="sidebar active"
>
  <div
    className="sidebar-info"
  >
    <UserAvatar />
    <UserInfo />
    <button
      className="info_more-btn"
      onClick={[Function hookProps.toggle]}
    >
      <span>
        Show Contacts 
      </span>
      <FaChevronDownIcon />
    </button>
  </div>
  <div
    className="sidebar-info_more"
  >
    <div
      className="separator"
    />
    <ContactList />
  </div>
</aside>"
`;
